receiver/k8slogreceiver/README.md

```diff
@@ -0,0 +1,173 @@
+# K8slog Receiver
+
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [development]: logs   |
+| Distributions | [] |
+| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Areceiver%2Fk8slog%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Areceiver%2Fk8slog) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Areceiver%2Fk8slog%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Areceiver%2Fk8slog) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@h0cheung](https://www.github.com/h0cheung), [@TylerHelmuth](https://www.github.com/TylerHelmuth) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+<!-- end autogenerated section -->
+
+Tails and parses logs in k8s environment.
+
+There only one mode of discovery as for now, it's specified by the `discovery.mode` configuration option:
+- `daemonset-stdout`: (default) Deployed as a DaemonSet, the receiver will read logs from the stdout of pods in the same node.
```

Comment:
Could we have a diagram for the 3 of them in the [design](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24439/files#diff-efa0f8665d59a187087f7f192e194b864b8be0a9cb6c181344b7bfe06a29587cR1) itself?

I'm interested to know if the `daemonset-stdout` mode will read the logs from the actual files or not. Could we clarify this in design document? If that's not the case I wonder if this should be default then cause I'm not sure if it could provide enough of delivery guarantees.

Specifically, the receiver should be capable to recover from where it was left in case of Collector's restarts. Reading from files gives us the option to keep track of the read offset (not sure if it's yet supported in filelog receiver, but it's sth that we would need anyways) while I'm not sure if reading from stdout can provide such a recovery guarantee.